### PR TITLE
[webview_flutter_android] Adds Android implementation to override console log

### PR DESCRIPTION
**Pull request by mvanbeusekom**
Adds the Android implementation for registering a JavaScript console callback. This will allow developers to receive JavaScript console messages in a Dart callback.

This PR contains the `webview_flutter_android` specific changes from PR #4541.

Related issue: flutter/flutter#32908

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/webview_flutter/webview_flutter_android/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 3.10.0
+
+* Adds support to register a callback to receive JavaScript console messages. See `AndroidWebViewController.setConsoleLogCallback`.
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_android/CHANGELOG.md:3]: nit:

```suggestion
* Adds support to register a callback to receive JavaScript console messages. See `AndroidWebViewController.onConsoleMessage`.
```

**Review by bparrishMines — APPROVED** [review]
LGTM with a nit

@stuartmorgan for secondary review

**Comment by mvanbeusekom** [comment]
@bparrishMines, took a long time to locate an error in the integration tests (accidentally kept a strong reference, where it should have been a weak reference). However that is fixed now and this PR should be ready for another review.

File: packages/webview_flutter/webview_flutter_android/pigeons/android_webview.dart [hunk 2]
```diff
@@ -57,6 +57,48 @@ enum FileChooserMode {
   save,
 }
 
+// TODO(bparrishMines): Enums need be wrapped in a data class because thay can't
+// be used as primitive arguments. See https://github.com/flutter/flutter/issues/87307
+class FileChooserModeEnumData {
+  late FileChooserMode value;
+}
+
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_android/pigeons/android_webview.dart:65]: It looks like this was added by accident. I couldn't find anywhere this is used.

```suggestion
```

**Review by bparrishMines — APPROVED** [review]
LGTM

**Pull request merged**